File: 2020/julia/p01.jl
const raw = readlines(open(joinpath(@__DIR__, "../input/01.txt")))
const data = parse.(Int, raw)

function ans1(data)
  nums = Set(data)
  for number in data
    other = 2020 - number
    if other ∈ nums 
      return other * number
    end
  end
end

println("Answer1: ", ans1(data))

function ans2(data)
  nums = Set(data)
  for (i, number) in enumerate(data)
    for number2 in data[i:end]
      remaining = 2020 - number - number2
      if remaining ∈ nums 
        return remaining * number * number2
      end
    end
  end
end

println("Answer2: ", ans2(data))

File: 2020/py/p01.py
from utils import data20

data = data20(1)


def answer1(inp):
  nums = set(int(x) for x in inp.splitlines())
  for x in nums:
    if (2020 - x) in nums:
      return x * (2020 - x)


def answer2(inp):
  nums = set(int(x) for x in inp.splitlines())
  for x in nums:
    for y in nums:
      if (2020 - x - y) in nums:
        return x * y * (2020 - x - y)


if __name__ == "__main__":
  ans1 = answer1(data)
  print("Answer1:", ans1)

  ans2 = answer2(data)
  print("Answer2:", ans2)
